db/collection.go

```diff
@@ -264,6 +265,12 @@ func (db *db) updateCollection(
 		return nil, err
 	}
 
+	schemaVersionHistoryKey := core.NewSchemaHistoryKey(desc.Schema.SchemaID, schemaVersionID)
+	err = txn.Systemstore().Put(ctx, schemaVersionHistoryKey.ToDS(), []byte(priorSchemaVersionID))
```

Comment:
question: This is somewhat confusing. `SchemaHistoryKey` is defined with `SchemaID` and `PriorSchemaVersion`, but here you are building the `NewSchemaHistoryKey` with the *current* `schemaVersionID`, and then using this key to point to a value, and that value is the `priorSchemaVersionID`. 

Feels like there is some inconsistent naming going on. Feel free to correct me.

Reply:
SchemaHistoryKey is an index of the next version, v1 points to v2 etc. This is documented on the key, and I cannot think of a better name. Am very open to alternatives though.